repair_picking_after_done/wizards/repair_move_transfer.py

```diff
@@ -73,7 +73,7 @@ def action_create_transfer(self):
         )
         picking.action_assign()
         if self.repair_order_id.lot_id:
-            stock_move.move_line_ids[0]._assign_production_lot(
-                self.repair_order_id.lot_id
+            stock_move.move_line_ids[0].write(
+                {"lot_id": self.repair_order_id.lot_id.id}
```

Comment:
Seems that Odoo did not cleanly remove this method: https://github.com/odoo/odoo/blob/16.0/addons/product_expiry/models/stock_move_line.py#L63 :speak_no_evil: